### ISSUE TITLE
reevaluate not working with replaceChild()

### ISSUE THREAD

**Issue by MikesBarto2002** [scottjehl/picturefill#685]
I am using the NPM version of Picturefill 3.0.2. What I need to do is replace an image that is poorly formatted with an image that is correctly formatted for responsive images. The content is generated dynamically as the user scrolls, hence the use of `reevaluate`.

I can't seem to get it working, though. When I call the picturefill function, it immediately drops me in the `noop()` call, and exits out of picturefill. I'm wondering if it has anything to do with using `replaceChild()`, or if I need to be using the mutation plugin. Any advice would be appreciated.

The code basically functions like this. I left out some of the code that is not important:

```
import picturefill from "picturefill";

export default function Images(element, type) {

    const images = {};

    /**
     * Converts the old image format in the DOM to the new 
     * figure tag with responsive images.
     * @param  {object} imageData The new image’s dataset.
     * @return {DOM} The new image in it’s figure tag.
     */
    images.buildInArticleImages = imageData => {

    	if (!imageData || !imageData.url) {
    		return null;
    	}

    	const imageElem = imageData.elem;

    	// Create figure tag
        const figureTag = document.createElement('figure');
        figureTag.id = `photo-${imageData.id}`;

        // Create image wrapper
        const figureWrapper = document.createElement('div');

        // Create link
        const link = document.createElement('a');
        link.href = imageData.url;

        // Create image
        const image = document.createElement('img');
        image.srcset = imageData.srcset.sources.toString();
        image.sizes = imageData.sizes;
        image.src = imageData.srcset.default;
        image.alt = imageData.caption;

        // Create caption
        let figcaption;
        if (imageData.caption) {
            figcaption = document.createElement('figcaption');
            figcaption.innerHTML = imageData.caption;
        } else {
            figcaption = null;
        }

        // Create credit
        let cite;
        if (
            imageData.credit &&
            imageData.credit !== ''
        ) {
            cite = document.createElement('cite');
            cite.innerHTML = `\xa0(${imageData.credit})`;
        }

        // Append elements to DOM
        if (link) {
            link.appendChild(image);
            figureWrapper.appendChild(link);
        } else {
            figureWrapper.appendChild(image);
        }
        if (figcaption) {
            if (cite) {
                figcaption.appendChild(cite);
            }
            figureWrapper.appendChild(figcaption);
        } else if (cite) {
            figureWrapper.appendChild(cite);
        }
        figureTag.appendChild(figureWrapper);

        // Replace image with figureTag
        imageElem.parentNode.replaceChild(figureTag, imageElem);

        return figureTag;

    };

    images.initImages = (() => {

    	// Grab all of the image data for the specific article
    	const imageData = images.getImageData();
    	// Build and replace each image on the page with the newly formatted image
    	if (imageData) {
	    const createImages = imageData.map(currentImage => {
	        return images.buildInArticleImages(currentImage);
	    };
	    // Refresh images with Picturefill
	    picturefill({
	        reevaluate: true,
	        elements: createImages
	    });
        }

    })();

    return images;

}
```

Basically, we grab the data for all images in a function called `getImageData()`. We then loop through each image, creating the new responsive image, and replacing the old image with this image.

**Comment by aFarkas** [comment]
I assume that it has to do with the usage of IDL attributes. These script-y things are polyfilled by the mutation plugin. You can therefore either use the mutation plugin and also drop the `picturefill` call entirely or use content attributes instead.

Please close if this helped.

**Comment by MikesBarto2002** [comment]
@aFarkas : I'm not sure I understand what you mean about content attributes. So does this mean changing all attributes being set from this:

`figureTag.id = 'someValue';`

To this:

`figureTag.setAttribute('id', 'someValue'):`

???

**Comment by aFarkas** [comment]
> Is using the mutation plugin as simple as dropping it on the page?

Yes, just include it together with the `picturefill` script and it will work.

**Comment by MikesBarto2002** [comment]
@aFarkas So I have async script injected both picturefill.min.js and pf.mutation.min.js onto my page. When I place a breakpoint on my picturefill call, it jumps straight to this line in picturefill:

`var noop = function() {};`

I made sure that I was passing an array-like object of elements into Picturefill as well. Anything else you might be able to suggest that I try?

**Comment by aFarkas** [comment]
In chrome picturefill is a noop. So everything looks fine there.
